docs/OSC-CONTEXT.md

```diff
@@ -0,0 +1,339 @@
+and the previously active context becomes its parent (if there is one). Only
+one context is currently active, but previously opened contexts remain valid in
+the background. Any other data written or read should be considered associated
+with the currently active context.
+
+Each context carries an identifier, chosen by the component opening the
+context. The identifier can chosen freely, but must not be longer than 64
+characters. The characters may be in the 32…126 byte range. Identifiers should
+be universally unique, for example randomly generated. A freshly generated UUID
+would work well for this, but this could also be something like the Linux boot
+ID combined with the 64bit inode number of Linux pidfds, or something hashed
+from it.
+
+Fundamentally, there are two OSC 3008 commands defined:
+
+1. OSC "`3008;start=`" … (the *start sequence*) → this initiates, updates or
+   indicates a return to a context. It carries a context identifier, and
+   typically some metadata. This may be sent to first initiate a context. If
+   sent again for the same context ID that was initiated already this indicates
+   an update of the existing context. In this case, *any* previously set
+   metadata fields for the context are flushed out, reset to their defaults,
+   and then reinitialized from the newly supplied data. Also, in this case any
+   subcontexts of the contexts are implicitly terminated.
+
+2. OSC "`3008;end=`" … (the *end sequence*)→ this terminates a context. It
+   carries a context identifier to close, initiated before with OSC
+   "`3008;start=`". It may also carry additional metadata.
+
+## General Syntax
+
+This builds on ECMA-48, and reuses the OSC and ST concepts introduced there.
+
+For sequences following this specification it is recommended to encode OSC as
+0x1B 0x5D, and ST as 0x1B 0x5C.
+
+ECMA-48 only allows characters from the range 0x20…0x7e (i.e. 32…126) inside
+OSC sequences. However, most terminal emulators nowadays allow the ASCII byte
+range > 0x7f in the OSC sequences they process, and so does this
+specification. Control characters (< 0x20 and 0x7f) are not allowed. The
+semicolon character ("`;`") – which is used as field separator by this
+specification – shall be replaced by "`\x3b`" and the backslash caracter
+("`\`") shall be replaced by "`\x5c`". All textual fields must be encoded in
+UTF-8, and then escaped with these two replacements.
+
+The start sequence begins with OSC, followed by the string `3008;start=`,
+followed by the context ID. This is then followed by any number of metadata
+fields, including none. Metadata fields begin with a semicolon (`;`) followed
+by in a string identifying the type of field, followed by an equal sign (`=`),
+and the field value. The sequence ends in ST.
+
+The end sequence begins with OSC, followed by the string `3008;end=`, followed
+by the context ID, and a series of metadata fields in the same syntax as for
+the start sequence.
```

Comment:
Maybe add "The sequence ends is ST." here as well. It is kind of obvious, but explicit is better in specifications IMHO.